client/my-sites/exporter/export-card/index.jsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import flowRight from 'lodash/flowRight';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SpinnerButton from './spinner-button';
+import FoldableCard from 'components/foldable-card';
+import analytics from 'lib/analytics';
+import Interval, { EVERY_SECOND } from 'lib/interval';
+import AdvancedSettings from './advanced-settings';
+import {
+	advancedSettingsFetch,
+	exportStatusFetch,
+	setPostType,
+	startExport,
+} from 'state/site-settings/exporter/actions';
+import {
+	shouldShowProgress,
+	getSelectedPostType,
+	isExporting,
+} from 'state/site-settings/exporter/selectors';
+
+class ExportCard extends Component {
+	componentWillMount() {
+		this.props.advancedSettingsFetch( this.props.siteId );
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.siteId !== this.props.siteId ) {
+			this.props.advancedSettingsFetch( newProps.siteId );
+		}
+	}
+
+	render() {
+		const {
+			translate,
+			siteId,
+		} = this.props;
+		const exportAll = () => this.props.startExport( siteId );
+		const exportSelectedItems = () => this.props.startExport( siteId, { exportAll: false } );
+		const fetchStatus = () => this.props.exportStatusFetch( siteId );
```

Comment:
I like this idea and agree the functions seemed superfluous, but went a slightly different direction. In 1449623 I moved the analytics recording out from the connect statement and into the component. Generally I prefer to keep the `connect` to the minimum required logic to connect what's needed with Redux. That way, anything non-Redux related isn't dependant on Redux logic and can live inside the component.

Btw, in the example above `siteId` won't actually be passed in via `ownProps`, as `connect` won't include props from the `mapStateToProps`. It's possible to add a third parameter to the `connect` statement (`mergeProps`) which combines the two, but generally it makes more sense to move the logic into the React component, or, as In this case, move the `siteId` to the parent component so that it does get passed in via `ownProps`.